backend/builder/tasks/task.py

```diff
@@ -21,45 +22,57 @@ class BuildTask(Task):
-
     def on_failure(
         self, exc: Exception, task_id: str, args: tuple, kwargs: dict[str, Any], einfo: ExceptionInfo
     ) -> None:
-        from django.db import close_old_connections
-
-        close_old_connections()
-
-        # channel_name, function, compute_task_key = self.split_args(args)
-
-        # failure_report = store_failure(exc, compute_task_key)
-        # error_type = compute_task_errors.get_error_type(exc)
-
-        # with get_orchestrator_client(channel_name) as client:
-        #     # On the backend, only execution errors lead to the creation of compute task failure report instances
-        #     # to store the execution logs.
-        #     if failure_report:
-        #         logs_address = {
-        #             "checksum": failure_report.logs_checksum,
-        #             "storage_address": failure_report.logs_address,
-        #         }
-        #     else:
-        #         logs_address = None
-
-        #     client.register_failure_report(
-        #         {"compute_task_key": compute_task_key, "error_type": error_type, "logs_address": logs_address}
-        #     )
-
-    def split_args(self, celery_args: tuple) -> tuple[str, orchestrator.Function, str]:
-        channel_name = celery_args[1]
-        compute_task_key = celery_args[2]
-        function = orchestrator.Function.parse_raw(celery_args[0])
-        return channel_name, function, compute_task_key
+        logger.error(exc)
+        logger.error(einfo)
+        function_key, channel_name = self.get_task_info(args, kwargs)
+        with get_orchestrator_client(channel_name) as client:
+            client.update_function_status(
+                function_key=function_key, action=orchestrator.function_pb2.FUNCTION_ACTION_FAILED
+            )
+
+    #  def on_failure(
+    #     self, exc: Exception, task_id: str, args: tuple, kwargs: dict[str, Any], einfo: ExceptionInfo
+    # ) -> None:
+    # super():on_failure(exc, task_id, args, kwargs, einfo)
+    # channel_name, function, compute_task_key = self.split_args(args)
+
+    # failure_report = store_failure(exc, compute_task_key)
+    # error_type = compute_task_errors.get_error_type(exc)
+
+    # with get_orchestrator_client(channel_name) as client:
+    #     # On the backend, only execution errors lead to the creation of compute task failure report instances
+    #     # to store the execution logs.
+    #     if failure_report:
+    #         logs_address = {
+    #             "checksum": failure_report.logs_checksum,
+    #             "storage_address": failure_report.logs_address,
+    #         }
+    #     else:
+    #         logs_address = None
+
+    #     client.register_failure_report(
+    #         {"compute_task_key": compute_task_key, "error_type": error_type, "logs_address": logs_address}
+    #     )
+
+    def before_start(self, task_id: str, args: tuple, kwargs: dict) -> None:
+        function_key, channel_name = self.get_task_info(args, kwargs)
```

Comment:
I feel like at least one of those should be `*args, **kwargs` 

If Celery imposing the signature, or is there another reason we do not control what parameters are passed?

Reply:
If not, I would be in favour of naming the parameters, it makes it easier to follow what is happening.

Reply:
[the celery signal pre_run (before_start)](https://docs.celeryq.dev/en/stable/userguide/signals.html#task-prerun) offers packed arguments, but if you see a more elegant solution, please let me know!

Reply:
I see... I guess there's no really better way to handle this, given that the arguments are already unpacked as soon as possible. Maybe worth a comment that it's a restriction from Celery, and that we cannot do more explicit?